netdcli/netdcli.go

```diff
@@ -373,46 +385,43 @@ func executeOpts(opts *cliOpts) error {
 
 	err := validateOpts(opts)
 	if err != nil {
-		log.Fatalf("error %v validating opts\n", err)
+		return err
 	}
 
-	etcdDriver := &state.EtcdStateDriver{}
-	driverConfig := &state.EtcdStateDriverConfig{}
-	driverConfig.Etcd.Machines = []string{opts.etcdURL}
-	config := &core.Config{V: driverConfig}
-	err = etcdDriver.Init(config)
+	stateDriver, err := utils.GetStateDriver()
```

Comment:
Ah ok. Sorry, I meant to say, -connect could take a free-form string and then the constructor could process that string. I do understand the argument about the defaults, but they could be passed if no arguments are relative to driver.

> On Jun 13, 2015, at 12:44 PM, Madhav Puri <notifications@github.com> wrote:
> 
> In netdcli/netdcli.go <https://github.com/contiv/netplugin/pull/99#discussion_r32373219>:
> 
> >  	}
> >  
> > -	etcdDriver := &state.EtcdStateDriver{}
> > -	driverConfig := &state.EtcdStateDriverConfig{}
> > -	driverConfig.Etcd.Machines = []string{opts.etcdURL}
> > -	config := &core.Config{V: driverConfig}
> > -	err = etcdDriver.Init(config)
> > +	stateDriver, err := utils.GetStateDriver()
> GetStateDriver() is just a utility to return a previously instantiated singleton state-driver (through a call to NewStateDriver()) so it doesn't need to take any arguments. I process the connection flags and init the state-driver instance in netdcli in the initStateDriver() function that is called just before executeOpts() and executeJson()
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/contiv/netplugin/pull/99/files#r32373219>.
>